flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/delegation/StreamPlanner.scala

```diff
@@ -190,10 +190,12 @@ class StreamPlanner(
     }
 
     new ExecNodeGraphInternalPlan(
-      JsonSerdeUtil
-        .createObjectWriter(ctx)
-        .withDefaultPrettyPrinter()
-        .writeValueAsString(execNodeGraph),
+      // ensures that the JSON output is always normalized
+      () =>
+        JsonSerdeUtil
+          .createObjectWriter(ctx)
```

Comment:
Should we rather freeze the `ObjectWriter` before passing it to the `Supplier`? Otherwise it's hard to tell what is the configuration used.

WDYT:
```
val writer = JsonSerdeUtil.createObjectWriter(ctx)
new ExecNodeGraphInternalPlan(
      // ensures that the JSON output is always normalized
      () =>  writer
        .withDefaultPrettyPrinter()
        .writeValueAsString(execNodeGraph),
      execNodeGraph)
)
```

Reply:
the `ctx` is already locked, also instantiating the writer might add some unnecessary overhead